### PR TITLE
Canham model add

### PR DESCRIPTION
**Pull request by Tess-LaCoil**
Added the Canham model material. I had to slightly change the testing code in order to include RK4 stan models that require a step size as argument.

All tests passed: https://github.com/traitecoevo/rmot/actions/runs/8610555357
have subsequently removed the branch from the testing workflow.

### CODE REVIEW

File: tests/testthat/fixtures/canham/make_canham.R [hunk 1]
```diff
@@ -0,0 +1,77 @@
+#Define specific functions for chosen DE, distribution of parameters, error process
+#Required library for multivariate normal
+require(mnormt)
```
Comment (fontikar) [inline tests/testthat/fixtures/canham/make_canham.R:3]: Can you add `mnormt` in the `DESCRIPTION` file under 'Suggests' and remove this `require` call please?

Comment (fontikar) [reply tests/testthat/fixtures/canham/make_canham.R:3]: Any functions from the `mnormt` package will need to be specified as `mnormt::funct()`

Comment (Tess-LaCoil) [reply tests/testthat/fixtures/canham/make_canham.R]: Done.

File: tests/testthat/helper-testing.R [hunk 2]
```diff
@@ -17,30 +17,47 @@ rmot_test_single_individual <- function(model_name,
...
+        rmot_run(chains = 1, iter = 1000, verbose = FALSE, show_messages = FALSE)
+    )
+  }
 
   # Extract samples and check if parameter estimates are reasonable.
   ind_samples <- rstan::extract(single_ind_test, permuted = TRUE,
                                          inc_warmup = FALSE)
   par_ests <- c()
-  for(i in length(par_names)){
+  for(i in 1:length(par_names)){
```
Comment (dfalster) [inline tests/testthat/helper-testing.R:51]: a safer option than `1:length(par_names)` is `seq_along(par_names)`. they differ when par_names has length 0.

File: tests/testthat/helper-testing.R [hunk 3]
```diff
@@ -49,19 +66,36 @@ rmot_test_single_individual <- function(model_name,
...
+        rmot_assign_data(step_size = data$step_size, #real
+                         n_obs = data$n_obs, #integer
+                         n_ind = data$n_ind, #integer
+                         y_obs = data$y_obs, #vector length N_obs
+                         obs_index = data$obs_index, #vector length N_obs
+                         time = data$time, #Vector length N_obs
+                         ind_id = data$ind_id, #Vector length N_obs
+                         y_0_obs = data$y_0_obs #vector length N_ind
+        ) |>
+        rmot_run(chains = 2, iter = 100, verbose = FALSE, show_messages = FALSE)
```
Comment (dfalster) [inline tests/testthat/helper-testing.R:82]: 2 chains?

File: tests/testthat/helper-testing.R [hunk 4]
```diff
@@ -49,19 +66,36 @@ rmot_test_single_individual <- function(model_name,
...
+      multi_ind_test <- rmot_model(paste0(model_name, "_multi_ind")) |>
+        rmot_assign_data(n_obs = data$n_obs, #integer
+                         n_ind = data$n_ind, #integer
+                         y_obs = data$y_obs, #vector length N_obs
+                         obs_index = data$obs_index, #vector length N_obs
+                         time = data$time, #Vector length N_obs
+                         ind_id = data$ind_id, #Vector length N_obs
+                         y_0_obs = data$y_0_obs #vector length N_ind
+        ) |>
+        rmot_run(chains = 2, iter = 100, verbose = FALSE, show_messages = FALSE)
```
Comment (dfalster) [inline tests/testthat/helper-testing.R:96]: 2 chains?

File: tests/testthat/test-rmot_models_canham.R [hunk 5]
```diff
@@ -0,0 +1,38 @@
+#Testing for canham model
+test_that("Model structures: canham", {
+  # Single individual
+  expect_named(rmot_model("canham_single_ind"))
```
Comment (dfalster) [inline tests/testthat/test-rmot_models_canham.R:4]: this should have actual names for comparison. 

e.g. if we have `z <- list(a=1, b=2)` then we'd run `expect_named(x, c("a", "b"))`

Also, why not have 
x <-rmot_model("canham_single_ind")
expect_named(x)
expect_type(x, "list")
expect_visible(x)

**Review by dfalster — APPROVED** [review]
Looks promising, some minor suggested changes.

I can't see any issues with stan files, but it's also end of the day!

File: tests/testthat/helper-testing.R (hunk 3 above)
Comment (Tess-LaCoil) [reply tests/testthat/helper-testing.R:82]: Yeah this is just testing that multiple chains produce the right size of samples.

File: tests/testthat/test-rmot_models_canham.R (hunk 5 above)
Comment (Tess-LaCoil) [reply tests/testthat/test-rmot_models_canham.R:4]: Done.

File: tests/testthat/helper-testing.R (hunk 2 above)
Comment (Tess-LaCoil) [reply tests/testthat/helper-testing.R:51]: Done.

**Pull request merged**